src/bin/maker-cli.rs

```diff
@@ -1,13 +1,14 @@
 use clap::Parser;
+use serde::{Deserialize, Serialize};
+use tokio::{io::BufReader, net::TcpStream};
+
 use coinswap::{
     maker::{
         error::MakerError,
         rpc::{read_rpc_message, RpcMsgReq},
     },
     utill::{send_message, setup_logger},
 };
-use serde::{Deserialize, Serialize};
-use tokio::{io::BufReader, net::TcpStream};
```

Comment:
Apologies for changes like these, this is the result of the `formatting` and `optimize import` done by RustRover IDE. If this needs to be reverted, please let me know.

Reply:
You can do a `cargo +nightly fmt --check` to check for the formatting errors, and to fix them, just do `cargo +nightly fmt`

Reply:
followed it, and pushed in https://github.com/citadel-tech/coinswap/commit/ac6e4fc197edecd0d4752c444adf1a6de151bee5

still there are some disparities.

Reply:
Because this is an unconditional recursion which isn't allowed by default. You may want to replace the recursion with a loop.
```rs
pub fn get_new_address(&self) -> Result<Address, WalletError> {
        let new_address = self.get_new_address()?;
        Ok(new_address)
}
```